panel/src/panel/upload.js

```diff
@@ -167,11 +167,15 @@ export default (panel) => {
 			files = [...files];
 
 			// add all files to the list as enriched objects
-			this.files = files.map((file) => this.file(file));
+			files = files.map((file) => this.file(file));
+
+			// merge the new files with already selected files
+			this.files = [...this.files, ...files];
```

Comment:
I think we somehow need to detect duplicates here.

Reply:
What would you think is a good way to detect this? Comparing the `src` property?

<img width="713" alt="Screenshot 2023-06-19 at 19 44 03" src="https://github.com/getkirby/kirby/assets/3788865/958c9749-9b05-41e0-90e2-1594dd684832">

Reply:
LOL. Notice to myself – First step: check out changes. Second step: leave comment 🙈